impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultPluginXmlFactory.java

```diff
@@ -93,7 +93,7 @@ public void write(XmlWriterRequest<PluginDescriptor> request) throws XmlWriterEx
                 new PluginDescriptorStaxWriter().write(outputStream, content);
             } else {
                 try (OutputStream os = Files.newOutputStream(path)) {
```

Comment:
ignore or use: `Variable 'os' is never used`

- <img width="874" alt="image" src="https://github.com/user-attachments/assets/d6cfdf22-13d1-4f90-b179-84ed8ea65d28" />
- <img width="1076" alt="image" src="https://github.com/user-attachments/assets/e24c78ee-76df-4d0b-b4d2-65102b4c94cf" />

Reply:
this looks like a critical bug and easy to prevent: https://checkstyle.sourceforge.io/checks/coding/unusedlocalvariable.html

Reply:
You wanna try adding the rule check to the PR ?

Reply:
its a bug in checkstyle we already have this checker in place:
- https://github.com/checkstyle/checkstyle/issues/17036 

@romani

Reply:
What version of checkstyle is in use?
Please share link to config also

Reply:
> Please share link to config also

config is mystery `ATM` as it comes from cloud.

Will update on this. Thanks

Reply:
Congo is inherited from parent, but there is a way to override it and complement with the rules needed.  You need to analyze how check style is configured in the parent.

Reply:
We have this bug reproducible and tested already in 
- https://github.com/checkstyle/checkstyle/pull/17039 
  
So its nice to have this info, but not mandatory anymore.